tests/test_radial.cc

```diff
@@ -38,6 +38,108 @@ inline auto random_spherical_dataset(std::vector<Eigen::VectorXd> points,
   return RegressionDataset<Eigen::VectorXd>(points, targets);
 }
 
+template <typename T> class RadialCovarianceTester : public ::testing::Test {
+public:
+  T test_case;
+};
+
+using RadialTestCases =
+    ::testing::Types<Exponential<EuclideanDistance>,
+                     SquaredExponential<EuclideanDistance>,
+                     Matern32<EuclideanDistance>, Matern52<EuclideanDistance>>;
+TYPED_TEST_SUITE(RadialCovarianceTester, RadialTestCases);
+
+TYPED_TEST(RadialCovarianceTester, test_edge_cases) {
+  double sigma = NAN;
+  for (const auto &pair : this->test_case.get_params()) {
+    if (pair.first.find("sigma") != std::string::npos) {
+      sigma = pair.second.value;
+    }
+  }
+  // the same points should be fully correlated
+  EXPECT_EQ(this->test_case(M_PI, M_PI), sigma * sigma);
+  // extremely close points should be almost perfectly correlated
+  EXPECT_NEAR(this->test_case(M_PI, M_PI + 1e-16), this->test_case(M_PI, M_PI),
+              1e-8);
+  // extremely distant points should be independent
+  EXPECT_EQ(this->test_case(0., 1e32), 0.);
+}
+
+TYPED_TEST(RadialCovarianceTester, test_derive_length_scale) {
+
+  auto set_sigma_length_scale = [&](double sigma, double length_scale) {
```

Comment:
do these capture anything but `this`?  maybe explicitly capture `this` rather than make the reader search for what's captured?